src/cpp/configuration/ParticipantConfiguration.cpp

```diff
@@ -28,9 +28,10 @@ namespace ddsrouter {
 ParticipantConfiguration::ParticipantConfiguration(
         ParticipantId id) noexcept
     : id_(id)
-    , type_(ParticipantType::PARTICIPANT_TYPE_INVALID)
     , raw_configuration_(RawConfiguration())
 {
+    // Set type from id
+    set_type_();
```

Comment:
Update doxygen of this function.
Depending where is it used, we may throw an Exception as in the other ParticipantConfiguration constructor